concourse/steps/release.py

```diff
@@ -334,8 +343,51 @@ def _push_tag(tag):
         else:
             raise NotImplementedError
 
+        if self.publishing_policy is ReleaseCommitPublishingPolicy.TAG_AND_MERGE_BACK:
+            def create_merge_commit(head: git.types.Commit_ish):
+                merge_base = self.git_helper.repo.merge_base(
+                    head,
+                    self.context().release_commit,
+                )
+
+                self.git_helper.repo.index.merge_tree(
+                    rhs=self.context().release_commit,
+                    base=merge_base,
+                )
+
+                return self.git_helper.index_to_commit(
+                    message=f'Merge release-commit from tag {self.release_tag}',
+                    parent_commits=(
+                        head,
+                        self.context().release_commit,
+                    ),
+                )
+
+            try:
+                head = self.git_helper.fetch_head(
+                    f'refs/heads/{self.repository_branch}'
+                ) # this is safe as release-tag is created already
```

Comment:
I think this comment is slightly misleading. what we want/try to do is to reduce the risk for conflicts / create a merge-commit against current tgt-branch's head. it might improve readability if this code would be factored into a (local) function w/ an expressive name (alternatively, add a comment explaining what is tried to be done - and why it is okay (or so we think) to not fail release in case of errors)